### ISSUE TITLE
Passive declare option is not taken into account by automatic recovery

### ISSUE THREAD

**Issue by motymichaely** [ruby-amqp/bunny#337]
When the connection is configured with `recover_from_connection_close`  a PRECONDITION_FAILED error occurs after connection recovery:

```
Caught #<Bunny::PreconditionFailed: PRECONDITION_FAILED - inequivalent arg 'durable' for queue 'some_queue' in vhost 'some_vhost': received 'false' but current is 'true'> while redeclaring and registering some_queue!
```

**Comment by michaelklishin** [comment]
This is already addressed for exchange recovery in #422, need to check queue recovery.

**Comment by michaelklishin** [comment]
According to a few more added tests, this is addressed. There is no evidence in the original
report that it's due to the `:passive` option but I don't see why it wouldn't work as it's passed
along with all other options during re-declaration on recovery and `:no_declare` will avoid re-declaration entirely.
